File: task6b.cpp
#include <iostream>
#include <string>
#include <algorithm>

// Функция для проверки, является ли число палиндромом
bool is_palindrome(int number) {
    std::string original = std::to_string(number);
    std::string reversed = original;
    std::reverse(reversed.begin(), reversed.end());
    return original == reversed;
}

int main() {
    int num1 = 12321; // Пример числа
    int num2 = 12345; // Пример числа

    // Проверяем, является ли хотя бы одно из чисел палиндромом
    bool result1 = is_palindrome(num1);
    bool result2 = is_palindrome(num2);

    std::cout << "Число " << num1 << " является палиндромом: " << result1 << std::endl;
    std::cout << "Число " << num2 << " является палиндромом: " << result2 << std::endl;

    return 0;
}

File: task6b.py
def is_palindrome(number):
    return str(number) == str(number)[::-1]

# Тестирование функции с двумя числами
num1 = 12321
num2 = 12345

# Проверяем, является ли хотя бы одно из чисел палиндромом
result1 = is_palindrome(num1)
result2 = is_palindrome(num2)

print(f"Число {num1} является палиндромом: {result1}")
print(f"Число {num2} является палиндромом: {result2}")
